pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,190 @@
+
+        if len(devices) == 0:
+            return
+
+    # copy console update from grub2.py
+    def write_config_console(self, config):
+        log.info("systemd.py: write_config_console")
+        if not self.console:
+            return
+
+        console_arg = "console=%s" % self.console
+        if self.console_options:
+            console_arg += ",%s" % self.console_options
+        self.boot_args.add(console_arg)
+
+
+
+    def write_config(self):
+        log.info("systemd.py: write_config systemd start")
+        # So at this point, all this is pretty much redundant
+        self.write_config_console(None)
+
+        # just rewrite the loader.conf
+        #cfg_file = config_file(self)
+        config_path = "%s%s" % (conf.target.system_root, "/boot/efi/loader/loader.conf")
+        log.info("systemd.py: write_config systemd loader conf : %s ", config_path)
+
+        # outside of resetting the default, this file should probably be moved to
+        # stubby as well?
+        config = open(config_path, "w")
+        config.write("timeout 3\n")
+        # create a default entry? No need right now..
+        # config.write("default 4142287a09f64398a50ac387bf9e1fca-*\n")
+        config.write("#console-mode keep\n")
+        config.close()
+
+        # do the bls command line here?
+        # update /etc/kernel/cmdline
+        # should look something like "root=UUID=45b931b7-592a-46dc-9c33-d38d5901ec29 ro  resume=/dev/sda3"
+        config_path = "%s%s" % (conf.target.system_root, "/etc/kernel/cmdline")
+        log.info("systemd.py: write_config systemd commandline : %s ", config_path)
+        config = open(config_path, "w")
+        args = " ".join(list(self.boot_args._arguments))
+        log.info("systemd.py: systemd used boot args: %s ", args)
+        # should run this through blkid to get UUID... but what is going on with the above args...
+        args += " root=/dev/mapper/fedora_fedora-root"
+        config.write(args)
+        config.close()
+        # rather than creating a mess in python lets just
+        # write the options above, and run a script which will merge the
+        # boot cmdline (after stripping inst. and BOOT_) with the anaconda settings
+        rc = util.execWithRedirect(
+            "/usr/sbin/updateloaderentries.sh",
+            [" "],
+            root=conf.target.system_root
+
+        )
+        if rc:
+            raise BootLoaderError("failed to write boot loader configuration")
+    #
+    # installation
+    #
+
+    def installbootmgr(self, args=None):
+        log.info("systemd.py: install")
+        if args is None:
+            args = []
+
+        log.info("systemd.py: install systemd boot install (root=%s)", conf.target.system_root)
+
+        rc = util.execWithRedirect("bootctl", [ "install", "--esp-path=/boot/efi" ],
+                                   root=conf.target.system_root,
+                                   env_prune=['MALLOC_PERTURB_'])
```

Comment:
Ah, I see that you are installing sd-boot using bootctl install. I thought that then there's no need to manually create a loader.conf. At least I didn't have to do it when migrated my machine from grub2 to use sd-boot...

Reply:
Do you need to explicitly use `--esp-path=/boot/efi` ? I also didn't have to do it an somehow `bootctl install` figured out that `/boot/efi/` was the mount point for the ESP partition.

Reply:
IIRC it needs the --esp-path when the /boot and /efi (or was it the efi variables) aren't setup yet. This bit predates the stubby package, which places the entries.srel file needed to get bootctl to place the loader files at the correct location. 

I will retest this to verify that behavior.

Reply:
Well, it turns out that it works fine without the --esp-path=, but the rest of anaconda (probably the packaging, too) can't deal with a /efi as the mount point, so I'm split about removing this vs being explicit about where we want it to install.

I think at the moment (as I prepare a new push) that i'm going to leave it in place.